### PR TITLE
Fixed `gardener-e2e-kind-ha-single-zone` failing tests.

### PR DESCRIPTION
**Pull request by seshachalam-yv**
**How to categorize this PR?**
/area testing
/kind test

**What this PR does / why we need it**:
This PR fixes [ci-gardener-e2e-kind-ha-single-zone prow job ](https://prow.gardener.cloud/view/gs/gardener-prow/logs/ci-gardener-e2e-kind-ha-single-zone/1579834505412218880) two failing tests. 
Recently this [PR](https://github.com/gardener/gardener/pull/6719) introduced e2e test to support HA test `ci-gardener-e2e-kind-ha-single-zone` 
1. `Shoot Tests Create Shoot, Rotate Credentials and Delete Shoot [Shoot, default, credentials-rotation]` test is failing, due to host not resolved.
``` log
Get \"https://gu-local--e2e-rotate.ingress.local-ha.seed.local.gardener.cloud:8448/\": dial tcp: lookup gu-local--e2e-rotate.ingress.local-ha.seed.local.gardener.cloud on 100.64.0.10:53: no such host\noccurred",
```
**Fix** : Based  on `HA_MODE`, using appropriate  seed name to update `ingress` host to `/etcd/hosts` file.
    
2. E2e ManagedSeed test is failing, local registry host are not resolved and referred wrong cluster name `gardener-local-control-plane` instead of `gardener-local-ha-control-plane`.   

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/provider-local/webhook/controlplane/ensurer.go [hunk 1]
```diff
@@ -45,27 +46,34 @@ func (e *ensurer) EnsureKubeletConfiguration(_ context.Context, _ gcontext.Garde
 }
 
 // EnsureAdditionalFiles ensures that additional required system files are added.
-func (e *ensurer) EnsureAdditionalFiles(_ context.Context, _ gcontext.GardenContext, new, _ *[]extensionsv1alpha1.File) error {
+func (e *ensurer) EnsureAdditionalFiles(ctx context.Context, gc gcontext.GardenContext, new, _ *[]extensionsv1alpha1.File) error {
+	clusterName := "gardener-local-control-plane"
+	cluster, err := gc.GetCluster(ctx)
```
Comment (acumino) [inline pkg/provider-local/webhook/controlplane/ensurer.go:50]: This error should not be ignored.

Comment (seshachalam-yv) [reply pkg/provider-local/webhook/controlplane/ensurer.go:50]: Fixed

**Review by acumino — COMMENTED** [review]
/lgtm

File: pkg/provider-local/webhook/controlplane/ensurer.go (hunk 1 above)
Comment (shafeeqes) [reply pkg/provider-local/webhook/controlplane/ensurer.go:51]: Isn't it better to return the error here?

File: pkg/provider-local/webhook/controlplane/ensurer.go [hunk 2]
```diff
@@ -45,27 +46,35 @@ func (e *ensurer) EnsureKubeletConfiguration(_ context.Context, _ gcontext.Garde
...
+	if err != nil {
+		return err
+	}
+	if gardencorev1beta1helper.IsHAControlPlaneConfigured(cluster.Shoot) {
+		clusterName = "gardener-local-ha-control-plane"
+	}
```
Comment (rfranzke) [inline pkg/provider-local/webhook/controlplane/ensurer.go]:
```suggestion
	cluster, err := gc.GetCluster(ctx)
	if err != nil {
		return err
	}
	
	kindClusterName := "gardener-local-control-plane"
	if gardencorev1beta1helper.IsHAControlPlaneConfigured(cluster.Shoot) {
		kindClusterName = "gardener-local-ha-control-plane"
	}

```

**Pull request merged**